src/vmm/src/lib.rs

```diff
@@ -301,6 +313,36 @@ impl VMM {
         Ok(())
     }
 
+    fn add_block_device(&mut self, cfg: &BlockConfig) -> Result<()> {
+        let mem = Arc::new(self.guest_memory.clone());
+
+        let range = MmioRange::new(MmioAddress(MMIO_MEM_START), 0x1000).unwrap();
+        let mmio_cfg = MmioConfig { range, gsi: 5 };
+
+        let mut guard = self.device_mgr.lock().unwrap();
+
+        let common = CommonArgs {
+            mem,
+            vm_fd: self.vm.vm_fd(),
+            event_mgr: &mut self.event_mgr,
+            mmio_mgr: guard.deref_mut(),
+            mmio_cfg,
+            kernel_cmdline: &mut self.kernel_cfg.cmdline,
+        };
+
+        let args = BlockArgs {
+            common,
+            file_path: PathBuf::from(&cfg.path),
+            read_only: false,
+            root_device: true,
+        };
+
+        // We can also hold this somewhere if we need to keep the handle for later.
```

Comment:
Even though we are no longer interacting with the block device after we create it, it would be nice to keep the handle in the VMM so that we can write more meaningful tests.

Reply:
Added a `block_devices` member to `VMM`. Is that similar to what you had in mind?

Reply:
Yes.

Reply:
Actually, I've looked a bit more at this and it looks like we only support one device right now. Moreover, calling the `add_block_device` twice would yield an error because the `MmioConfig` is hard coded. How about we actually use Option<Block> instead of a vector for now? And also maybe check if the self.block is None, otherwise return an error?